### ISSUE TITLE
`llvm-readobj` not found

### ISSUE THREAD

**Issue by Jasper-Bekkers** [rust-mobile/xbuild#129]
```
Caused by:
    0: Unable to read needed libraries from `....so`
    1: Failed to run `"llvm-readobj" "--needed-libs" "....so"`
    2: program not found
```

It appears that even though xbuild installs (some parts) of the NDK, it ultimately fails to find `llvm-readobj` in this installation causing clean builds to fail on windows.

**Comment by Jasper-Bekkers** [comment]
Hey @dvc94ch I don't quite understand why you changed my issue?

**Comment by dvc94ch** [comment]
Well, we don't need an issue for every missing package, one to track missing packages is enough

**Comment by dvc94ch** [comment]
They're actually not even missing, as we test windows in ci, but I understand that in your organization not everyone can install standard tools. But that doesn't mean we should add feature creep to xbuild. Maybe you can prepare a windows disk image for your employees?

**Comment by dvc94ch** [comment]
Or some active directory deployment, I'm sure windows has plenty of tools to preinstall stuff and prevent employees from messing up the installation

**Comment by duckinator** [comment]
[Due to there being no appropriate flag on the installer](https://github.com/llvm/llvm-project/issues/54724), you can't just do `winget install LLVM.LLVM` and get it in your PATH.

However, until that's resolved: `winget install -e -i --id LLVM.LLVM` will show the GUI for the installer, which lets you select the option to add it to the PATH.

This gets you all of the LLVM tools.

---

I have not tested it, but it looks like winget also has `Rustlang.Rustup`, which (if it works properly) should address the Rust toolchain.

**Comment by dvc94ch** [comment]
Some docs would be great! I don't have a windows machine and the GitHub ci has chocolatey pre installed. But I know @MarijnS95 had some issues with installing all packages with winget

**Comment by duckinator** [comment]
It's worth someone trying this in a fresh Windows VM to make sure it actually works, but this _may_ be all you need for Android development using a pure-Rust codebase:

```
winget install -e -i --id LLVM.LLVM # Make sure to enable the option to add it to your PATH.
winget install Rustlang.Rustup Google.AndroidStudio
```

I don't see a way to install the iOS tools, `mksquashfs`, `kotlin`, or `gradle` via winget.

Things to be determined:
- do any of the Java packages on winget work for this? if so, which one?
- does `Google.AndroidStudio` include `adb`? if not, is there a different way to get it via winget?
- does `Rustlang.Rustup` actually work well enough to be used for this?

**Comment by MarijnS95** [comment]
@duckinator that's what we have in our own documentation, but having `-i` in the `winget` command and requiring developers to manually click the right options in the visual installer is cumbersome.

I've been following https://github.com/microsoft/winget-pkgs/issues/11767 and https://github.com/llvm/llvm-project/issues/54724 for a long time, but nothing appears to be moving :(.

Not that it matters much, `xbuild` throws a stripped NDK in a user path, which `cc-rs`' `--sysroot` fails on when the username includes spaces (https://github.com/rust-mobile/xbuild/issues/124). The only "workaround" I've considered is not stripping `clang` out of the NDK prebuilts, or perhaps there's a user-writable (not requiring root) path outside of `%USERPROFILE%` that is known to never have spaces, which `xbuild` could install to?

Until that's resolved, `cargo-apk` is still the go-to solution which does not suffer from this issuee.

**Comment by duckinator** [comment]
@MarijnS95 I agree, it's pretty cumbersome and frustrating. I'm just trying to document the state of things in a public spot (to hopefully save other folks the hassle of figuring it out themselves like I did).
